tests/integration_tests/tests/agent_tests/test_plugin_update.py

```diff
@@ -274,23 +269,31 @@ def _perform_update(self, update_plugins=True):
         self.wait_for_execution_to_end(execution)
 
     def _get_work_dir_for_plugin(self):
-        return os.path.join(self.workdir, self.plugin_name)
+        return str(self.workdir / self.plugin_name)
 
     def _get_source_dir_for_plugin(self):
-        plugin_dir = '{0}'.format(self.plugin_name)
-        return resource(os.path.join('plugins', plugin_dir))
+        plugins_dir = os.path.dirname(integration_tests_plugins.__file__)
+        return os.path.join(plugins_dir, self.plugin_name)
 
     def _copy_plugin_files_to_work_dir(self):
         source_dir = self._get_source_dir_for_plugin()
         target_dir = self._get_work_dir_for_plugin()
-        shutil.copytree(source_dir, target_dir)
+        if not os.path.exists(target_dir):
+            shutil.copytree(source_dir, target_dir)
         return target_dir
 
     def _upload_v_2_plugin(self):
-        plugin_work_dir = self._copy_plugin_files_to_work_dir()
-
-        self._replace_version(plugin_work_dir, *self.versions)
-        self.upload_mock_plugin(self.plugin_name, plugin_path=plugin_work_dir)
+        plugin_name = 'version_aware_v2'
+        plugins_dir = os.path.dirname(integration_tests_plugins.__file__)
+        plugin_source = os.path.join(plugins_dir, plugin_name)
+
+        wagon_paths = self._get_or_create_wagon(plugin_source)
+        for wagon_path in wagon_paths:
+            yaml_path = os.path.join(plugin_source, 'plugin.yaml')
+            with zip_files([wagon_path, yaml_path]) as zip_path:
+                self.client.plugins.upload(zip_path, visibility='global')
+        self.logger.info(
+            'Finished uploading {0}...'.format(plugin_name))
```

Comment:
what's all this, we already have the fixture

Reply:
In _test_single_deployment_is_updated_ and _test_many_deployments_are_updated_ I need to upload version 2 of the plugin **after** the BP is installed. Don't see how can I use the fixture for that